# 读取数据

In [2]:
import pandas as pd
import numpy as np

In [3]:
df = pd.read_csv("D:/VsCodeProjects/Ecommerce_User_Analysis/data/sample.csv")

In [4]:
df.head()

,user_id,item_id,category_id,behavior_type,timestamp
0,128078,4630672,2033408,pv,1512266009
1,10035,3667432,1080785,pv,1511890180
2,109218,2403812,3430419,pv,1512232651
3,106205,5150006,1165503,pv,1511855219
4,1002307,905583,2520377,pv,1511795200


In [5]:
df.shape

(2003016, 5)

# 数据概览

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2003016 entries, 0 to 2003015
Data columns (total 5 columns):
 #   Column         Dtype 
---  ------         ----- 
 0   user_id        int64 
 1   item_id        int64 
 2   category_id    int64 
 3   behavior_type  object
 4   timestamp      int64 
dtypes: int64(4), object(1)
memory usage: 76.4+ MB


In [7]:
df.describe()

,user_id,item_id,category_id,timestamp
count,2.003016e+06,2.003016e+06,2.003016e+06,2.003016e+06
mean,5.070216e+05,2.577960e+06,2.697180e+06,1.511949e+09
std,2.940243e+05,1.488589e+06,1.463421e+06,5.997935e+06
min,1.000000e+00,1.900000e+01,2.171000e+03,-2.034494e+09
25%,2.525478e+05,1.291632e+06,1.320293e+06,1.511762e+09
50%,5.040775e+05,2.577764e+06,2.672762e+06,1.511964e+09
75%,7.610290e+05,3.859871e+06,4.145813e+06,1.512179e+09
max,1.018011e+06,5.163064e+06,5.161669e+06,1.761418e+09


# 缺失值检查

In [8]:
df.isnull().sum()

user_id          0
item_id          0
category_id      0
behavior_type    0
timestamp        0
dtype: int64

# 重复值检查

In [9]:
df.duplicated().sum()

np.int64(0)

# 时间戳转换

In [10]:
df["datetime"] = pd.to_datetime(df["timestamp"], unit="s") + pd.Timedelta(hours=8)

In [11]:
df[["timestamp", "datetime"]].head()

,timestamp,datetime
0,1512266009,2017-12-03 09:53:29
1,1511890180,2017-11-29 01:29:40
2,1512232651,2017-12-03 00:37:31
3,1511855219,2017-11-28 15:46:59
4,1511795200,2017-11-27 23:06:40


# 提取时间特征

In [12]:
df["date"] = df["datetime"].dt.date

In [13]:
df["hour"] = df["datetime"].dt.hour

In [14]:
df["weekday"] = df["datetime"].dt.day_name()

In [15]:
df.head()

,user_id,item_id,category_id,behavior_type,timestamp,datetime,date,hour,weekday
0,128078,4630672,2033408,pv,1512266009,2017-12-03 09:53:29,2017-12-03,9,Sunday
1,10035,3667432,1080785,pv,1511890180,2017-11-29 01:29:40,2017-11-29,1,Wednesday
2,109218,2403812,3430419,pv,1512232651,2017-12-03 00:37:31,2017-12-03,0,Sunday
3,106205,5150006,1165503,pv,1511855219,2017-11-28 15:46:59,2017-11-28,15,Tuesday
4,1002307,905583,2520377,pv,1511795200,2017-11-27 23:06:40,2017-11-27,23,Monday


# 检查行为类型

In [16]:
df["behavior_type"].value_counts()

behavior_type
pv      1794141
cart     110316
fav       58098
buy       40461
Name: count, dtype: int64

# 保存清洗后数据

In [17]:
df.to_csv("D:/VsCodeProjects/Ecommerce_User_Analysis/data/cleaned_data.csv", index=False)

## 数据清洗结论

1. 基于固定随机种子对原始数据进行 2% 抽样，共获得 2,003,016 条用户行为记录；
2. 数据不存在缺失值，完整性较高；
3. 数据不存在完全重复记录，无需去重；
4. 将 Unix 时间戳转换为标准时间格式，并提取日期、小时及星期等时间特征；
5. 数据覆盖 2017 年 11 月 25 日至 2017 年 12 月 3 日的用户行为，可支撑后续用户行为分析、漏斗分析及 RFM 用户价值分析。
### 时间字段处理说明

原始数据中的 Unix 时间戳采用 UTC 标准时间记录。考虑到淘宝平台用户主要来自中国大陆，本项目将时间统一转换为 UTC+8（北京时间），并基于修正后的时间提取小时、日期及星期等特征，以保证后续行为分析结果符合实际业务场景。